cmd/qgb/helpers/parse.go

```diff
@@ -0,0 +1,20 @@
+package helpers
+
+import (
+	"github.com/libp2p/go-libp2p/core/peer"
+	tmlog "github.com/tendermint/tendermint/libs/log"
+)
+
+// ParseAddrInfos converts strings to AddrInfos
+func ParseAddrInfos(logger tmlog.Logger, addrs []string) ([]peer.AddrInfo, error) {
+	infos := make([]peer.AddrInfo, 0, len(addrs))
+	for _, addr := range addrs {
+		info, err := peer.AddrInfoFromString(addr)
+		if err != nil {
+			logger.Error("parsing info from multiaddr", "addr", addr, "err", err)
+			return nil, err
+		}
+		infos = append(infos, *info)
```

Comment:
[non blocking question]
dereferencing this value makes sense to me, as I never like using pointers when we don't strictly need to (like for a read only struct such as `AddrInfo`), especially since its already a slice. However, I'm uncertain if there is a reason why we need a pointer, since the maintainers return a pointer. Do we know why they use a pointer?

Reply:
I took the implementation from Celestia-node :cat:

I guess the reason for using pointers is that the addresses could get heavy and we don't want to copy them every time. 

What dl you think? Should i switch to using pointers?

Reply:
I took the implementation from Celestia-node :cat:

I guess the reason for using pointers is that the addresses could get heavy and we don't want to copy them every time. 

What dl you think? Should i switch to using pointers?

Reply:
nah, if celestia-node uses it and they don't have issues its probably fine. It should be an improvement to not use pointers, as it makes no sense to me, but I'm just paranoid lol.

Reply:
Shit, after merging I remembered that I did couple changes, here is the original implementation:
https://github.com/celestiaorg/celestia-node/blob/bd96b3dab06e7d733cf0e62adb94802e2007a407/nodebuilder/p2p/bootstrap.go#L45-L64

But internally, it's almost the same functions called. Do I revert? or we keep it until we have an issue?